lab_1/scripts/update-pipeline-definition.sh

```diff
@@ -0,0 +1,125 @@
+newPipelineJson="./pipeline-$(date +'%m-%d-%Y').json"
+pipelineJson=./pipeline.json
+
+# flags default values
+branch="main"
+owner="$(git config user.name)"
+pollForSourceChanges=false
+repo="$(basename `git rev-parse --show-toplevel`)"
+
+# Check if jq installed
+type jq >/dev/null 2>&1
+jqInstalledCode=$?
+
+errorTitle="\033[0;31m Error.\033[0m"
+
+if [ "$jqInstalledCode" -ne 0 ]; then
+    echo "$errorTitle Jq not found. Install Jq to proceed. See https://stedolan.github.io/jq/";
+
+    echo "    Ubuntu Installation: sudo apt install jq"
+    echo "    MacOS Installation: brew install jq"
+
+    exit;
+fi
+
+
+firstArg=$1;
+argsQty=$#;
+
+# Read flags
+while [[ $# -gt 0 ]]
+    do key="$1"
+    case $key in
+        --branch)
+            branch="$2"
+            shift;;
+        --owner)
+            owner="$2"
+            shift;;
+        --poll-for-source-changes)
+            pollForSourceChanges="$2"
+            shift;;
+        --repo)
+            repo="$2"
+            shift;;
+        --configuration)
+            configuration="$2"
+            shift;;
+        *)
+            if [ "$firstArg" = "$key" ]; then
+                if [[ ! -f "$firstArg" ]]; then
+                    echo "$errorTitle file $firstArg doesn't exist. Specify correct pipeline name to continue"
+                    exit 1;
+                else
+                    pipelineJson=$key
+                fi
+            fi;;
+    esac
+    shift
+done
+
+cp $pipelineJson $newPipelineJson
+jqtmp=./tmp.$$.json
+
+# remove metadata
+jq 'del(.metadata)' $newPipelineJson > $jqtmp && mv $jqtmp $newPipelineJson
+
+# increment version
+jq '.pipeline.version = .pipeline.version + 1' $newPipelineJson > $jqtmp && mv $jqtmp $newPipelineJson
+
+
+# Perform additional actions if extra flags set
+if [[ $argsQty -gt 1 ]]; then
+    # Check necessary flags are set
+    if [[ ! $configuration ]]; then
+        echo "$errorTitle You didn't specify --configuration flag."
+        exit 1;
```

Comment:
The criteria says:
```
It should perform only 1.1 and 1.2 actions if no additional parameters are provided. E.g.: $ ./update-pipeline-definition.sh ./pipeline.json.
```

If  `configuration` is not specified, but `owner` is specified, you should update only the values you were given, instead of exiting the entire script